src/Core/Ripgrep.re

```diff
@@ -214,9 +216,20 @@ let search = (~executablePath, ~directory, ~onUpdate, ~onComplete) => {
     });
   };
 
+  let globs =
+    filesExclude
+    |> List.map(x => "!" ++ x)
+    |> List.map(x => ["-g", x])
+    |> List.concat;
+
+  let args =
+    globs
+    @ ["--smart-case", "--hidden", "--files", "--", directory]
+    |> Array.of_list;
+
```

Comment:
This is the first time we apply user input as arguments to flags. Do we need to sanitize or escape this to prevent "flag injection", @bryphe? What happens if `x` starts with whitespace? Should it be trimmed?

Reply:
Good question @glennsl . I believe this should be safe from a 'command injection' attack (where some rogue config would set an option that would allow us to escape out and execute arbitrary shell commands). 

The arguments we're passing are split out and separate from the executable command: https://github.com/revery-ui/rench/blob/a976fe51b89b4943e8650f8b9d6ebc49331678b0/src/ChildProcess.re#L126

(It would be more of a concern if we were passing the entire command-line to the shell).

> What happens if x starts with whitespace? Should it be trimmed?

I don't believe this is needed, but it would be worth testing (it could behave differently on different platforms). Worst case would be that the argument would not be processed as expected.

Thanks for reviewing and thinking about this!

Reply:
> I believe this should be safe from a 'command injection' attack (where some rogue config would set an option that would allow us to escape out and execute arbitrary shell commands).

Yeah I wasn't so much worried about that as being able to pass arbitrary flags to `rg`, accidentally or not.

Reply:
> Yeah I wasn't so much worried about that as being able to pass arbitrary flags to rg, accidentally or not.

Ah, got it. I believe when the arguments are passed as an array with `create_process_env`, it's like having single quotes around them (`rg '-gtest'`), so I'm pretty sure we're safe from arbitrary flags - the glob pattern may break though (ie, `rg '-g! -test'`)